networkx/algorithms/covering.py

```diff
@@ -0,0 +1,63 @@
+"""
+An edge cover of a graph is a set of edges such that every vertex of
+the graph is incident to at least one edge of the set.
+A minimum edge cover is an edge covering of smallest possible size.
+
+`Edge Cover <https://en.wikipedia.org/wiki/Edge_cover>`_
+"""
+#    Copyright (C) 2015 by
+#    Nishant Nikhil <nishantiam@gmail.com>
+#    All rights reserved.
+#    BSD license.
+
+import networkx as nx
+from networkx.utils import not_implemented_for
+
+
+@not_implemented_for('directed')
+@not_implemented_for('multigraph')
+def min_edge_cover(G):
+    """Returns a set of edges which constitute the minimum edge cover of the graph.
+    A smallest edge cover can be found in polynomial time by finding
+    a maximum matching and extending it greedily so that all vertices
+    are covered.
+
+    Parameters
+    ----------
+    G : An undirected, unweighted graph.
+
+    Returns
+    -------
+    min_cover : set
+        The covering is returned as a set, ``min_cover``.
+        It contains all the edges of minimum edge cover
+        in form of tuples.
+
+    Notes
+    -----
+    Worst Case runtime of ``nx.max_weight_matching(G, maxcardinality=True)``
+    is `O(number_of_nodes^{3})`
+    And the greedy extension works in `O(n\log{}n)`
+    So worst case run time of the function is `O(number_of_nodes^{3})`.
+    """
+    maximum_matching = {}
+    is_cover_exist = True
+    for node in G.nodes():
+        if not G[node]:
+            is_cover_exist = False
+            break
+    if not is_cover_exist:
```

Comment:
This would be more concise as `if all(d == 0 for n, d in G.degree()):`, or something similar.